internal/http/client.go

```diff
@@ -32,8 +32,15 @@ var (
 
 // NewRelicClient represents a client for communicating with the New Relic APIs.
 type NewRelicClient struct {
-	Client     *retryablehttp.Client
-	Config     config.Config
+	// Client represents the underlying HTTP client.
+	Client *retryablehttp.Client
+
+	// Config is the HTTP client configuration.
+	Config config.Config
+
+	// UsePersonalAPIKeyCompatability is for internal use only.
+	UsePersonalAPIKeyCompatibility bool
```

Comment:
Not sure that compatibility is the right item here, could be just `UsePersonalAPIKey`

Reply:
`PreferPersonalAPIKey`?  It won't be used if it's not passed